spec.html

```diff
@@ -472,16 +472,12 @@ <h1>ShadowRealm ( )</h1>
 			<emu-alg>
 				1. If NewTarget is *undefined*, throw a *TypeError* exception.
 				1. Let _O_ be ? OrdinaryCreateFromConstructor(NewTarget, *"%ShadowRealm.prototype%"*, « [[ShadowRealm]], [[ExecutionContext]] »).
-				1. Let _realmRec_ be CreateRealm().
-				1. Set _O_.[[ShadowRealm]] to _realmRec_.
-				1. Let _context_ be a new execution context.
-				1. Set the Function of _context_ to *null*.
-				1. Set the Realm of _context_ to _realmRec_.
-				1. Set the ScriptOrModule of _context_ to *null*.
+				1. Perform ? InitializeHostDefinedRealm().
+				1. Let _context_ be the running execution context.
```

Comment:
I wasn't aware of https://github.com/tc39/ecma262/pull/3139, but `InitializeHostDefinedRealm` doing the push of the context without restoring it seems like a very problematic pattern, and the name of the AO doesn't reveal that. I don't recall a prior art for such pattern either.